app/src/main/java/com/android/voyageur/ui/search/SearchUserProfile.kt

```diff
@@ -84,28 +84,31 @@ fun SearchUserProfileScreen(userViewModel: UserViewModel, navigationActions: Nav
 fun SearchUserProfileContent(userData: User, userViewModel: UserViewModel) {
   val currentUser by userViewModel.user.collectAsState()
   val sentFriendRequests by userViewModel.sentFriendRequests.collectAsState()
-
-  // Fetch sent friend requests and contacts when the composable is first displayed
-  LaunchedEffect(Unit) {
-    userViewModel.getSentFriendRequests()
-    userViewModel.getMyContacts {}
-  }
+  val receivedFriendRequests by userViewModel.friendRequests.collectAsState()
 
   val isContactAdded = currentUser?.contacts?.contains(userData.id) ?: false
   val isRequestPending = sentFriendRequests.any { it.to == userData.id }
+  val isRequestReceived = receivedFriendRequests.any { it.from == userData.id }
+  val friendRequest = receivedFriendRequests.find { it.from == userData.id }
   val signedInUserId = currentUser?.id ?: ""
 
   UserProfileContent(
       userData = userData,
       signedInUserId = signedInUserId,
       isContactAdded = isContactAdded,
       isRequestPending = isRequestPending,
+      isRequestReceived = isRequestReceived,
       onAddContact = { userViewModel.sendContactRequest(userData.id) },
       onRemoveContact = { userViewModel.removeContact(userData.id) },
       onCancelRequest = {
         val requestId = userViewModel.getSentRequestId(userData.id)
         if (requestId != null) {
           userViewModel.deleteFriendRequest(requestId)
         }
+      },
+      onAcceptRequest = {
+        if (friendRequest != null) {
```

Comment:
Why are we null checking here? If the friendRequest is null there shouldn't be an accept button right?

Reply:
Thank you for the feedback! While I understand your point that the Accept button should not be displayed if friendRequest is null, I believe the null check here is still necessary for robustness. 
Firstly, the UI logic (whether the button is displayed) and the backend logic (processing the request) should remain decoupled. This allows for more flexible and maintainable code. Even if the UI is correct now, there could be scenarios in the future where this assumption doesn't hold. 
Secondly, including the null check ensures the app doesn't run into unexpected crashes or undefined behaviour in edge cases where the friendRequest might be null due to synchronisation issues, a race condition, or a state update after the button was displayed.